packages/graphql-hooks/src/useClientRequest.js

```diff
@@ -103,6 +104,13 @@ function useClientRequest(query, initialOpts = {}) {
     }
 
     const revisedCacheKey = client.getCacheKey(revisedOperation, revisedOpts)
+    const stringifiedCacheKey = JSON.stringify(revisedCacheKey)
+
+    // NOTE: Up until this point all calls will be called in correct order
+    // but after network request there's possibility of race conditions so
+    // activeCacheKey will enshure state to be updatet only with last value
+    activeCacheKey.current = stringifiedCacheKey
```

Comment:
This can then become:

```suggestion
    activeCacheKey.current = revisedCacheKey
```